Sources/SwiftDocC/Model/ParametersAndReturnValidator.swift

```diff
@@ -0,0 +1,473 @@
+                return range
+            }
+        }
+        return range
+    }
+    
+    // MARK: Diagnostics
+    
+    private func extraReturnsProblem(_ returns: Return, symbolKind: SymbolGraph.Symbol.Kind?) -> Problem {
+        return Problem(
+            diagnostic: Diagnostic(
+                source: returns.range?.source,
+                severity: .warning,
+                range: adjusted(returns.range),
+                identifier: "org.swift.docc.VoidReturnDocumented",
+                summary: "Return value documented for \(symbolKind?.displayName.lowercased() ?? "symbol") returning void"
+            ),
+            possibleSolutions: [
+                Solution(
+                    summary: "Remove return value documentation",
+                    replacements: returns.range.map { [Replacement(range: adjusted($0), replacement: "")] } ?? []
+                )
+            ]
+        )
+    }
+    
+    private func extraParameterProblem(_ parameter: Parameter, knownParameterNames: Set<String>, symbolKind: SymbolGraph.Symbol.Kind?) -> Problem {
+        let source = parameter.range?.source
+        
+        let summary = "Parameter \(parameter.name.singleQuoted) not found in \(symbolKind.map { $0.displayName.lowercased() } ?? "the") declaration"
+        let identifier = "org.swift.docc.DocumentedParameterNotFound"
+        
+        let nearMisses = NearMiss.bestMatches(for: knownParameterNames, against: parameter.name)
+        
+        if nearMisses.isEmpty {
+            // If this parameter doesn't resemble any of this symbols parameters, suggest to remove it.
+            return Problem(
+                diagnostic: Diagnostic(source: source, severity: .warning, range: adjusted(parameter.range), identifier: identifier, summary: summary),
+                possibleSolutions: [
+                    Solution(
+                        summary: "Remove \(parameter.name.singleQuoted) parameter documentation",
+                        replacements: parameter.range.map { [Replacement(range: adjusted($0), replacement: "")] } ?? []
+                    )
+                ]
+            )
+        }
+        
+        // Otherwise, suggest to replace the documented parameter name with the one of the similarly named parameters.
+        return Problem(
+            diagnostic: Diagnostic(source: source, severity: .warning, range: adjusted(parameter.nameRange), identifier: identifier, summary: summary),
+            possibleSolutions: nearMisses.map { candidate in
+                Solution(
+                    summary: "Replace \(parameter.name.singleQuoted) with \(candidate.singleQuoted)",
+                    replacements: parameter.nameRange.map { [Replacement(range: adjusted($0), replacement: candidate)] } ?? []
+                )
+            }
+        )
+    }
+    
+    private func externalParameterNameProblem(_ parameter: Parameter, knownParameterNamesForArgumentLabel: [String]) -> Problem {
```

Comment:
This diagnostic is especially confusing. Could you put an example of this (or ideally all of these diagnostics) in code comments to make it simpler for DocC developers to understand what's happening here?